src/CtrlGroup.c

```diff
@@ -724,6 +724,22 @@ BOOL Ros_CtrlGroup_IsRobot(CtrlGroup* ctrlGroup)
     return((ctrlGroup->groupId >= MP_R1_GID) && (ctrlGroup->groupId <= MP_R8_GID));
 }
 
+//-------------------------------------------------------------------
+// Returns TRUE if the specified group is defined as a base
+//-------------------------------------------------------------------
+BOOL Ros_CtrlGroup_IsBase(CtrlGroup const* const ctrlGroup)
+{
+    return ((ctrlGroup->groupId >= MP_B1_GID) && (ctrlGroup->groupId <= MP_B8_GID));
```

Comment:
these duplicate base functionality, but we can't use those on all platforms, hence these custom implementations.

(they will need to be updated for #19 as those controllers support fewer groups)